bolt-boost/src/types.rs

```diff
@@ -113,40 +108,47 @@ impl TryFrom<ConstraintsMessage> for ConstraintsWithProofData {
 
 /// Calculate the SSZ hash tree root of a transaction, starting from its enveloped form.
 /// For type 3 transactions, the hash tree root of the inner transaction is taken (without blobs).
-fn calculate_tx_hash_tree_root(
-    envelope: &TxEnvelope,
-    raw_tx: &Bytes,
-) -> Result<B256, Eip2718Error> {
-    match envelope {
-        // For type 3 txs, take the hash tree root of the inner tx (EIP-4844)
-        TxEnvelope::Eip4844(tx) => match tx.tx() {
-            TxEip4844Variant::TxEip4844(tx) => {
-                let mut out = Vec::new();
-                out.put_u8(0x03);
-                tx.encode(&mut out);
-
-                Ok(tree_hash::TreeHash::tree_hash_root(&Transaction::<
-                    <DenebSpec as EthSpec>::MaxBytesPerTransaction,
-                >::from(out)))
-            }
-            TxEip4844Variant::TxEip4844WithSidecar(tx) => {
-                use alloy_rlp::Encodable;
-                let mut out = Vec::new();
-                out.put_u8(0x03);
-                tx.tx.encode(&mut out);
+fn calculate_tx_proof_data(raw_tx: &Bytes) -> Result<(TxHash, HashTreeRoot), Eip2718Error> {
+    let is_type_3 = raw_tx
+        .first()
+        .ok_or(Eip2718Error::RlpError(alloy_rlp::Error::Custom("empty RLP bytes")))?
+        == &0x03;
```

Comment:
maybe more readable, up to you
```suggestion
    let Some(is_type_3) = raw_tx.first().map(|type| type == &0x03) else {
    	return Eip2718Error::RlpError(alloy_rlp::Error::Custom("empty RLP bytes"));
    };
```

Reply:
I like it! Thanks!